src/AddressRegistry.Api.Backoffice/Address/AddressCrabEditClient.cs

```diff
@@ -0,0 +1,199 @@
+                    Position = MapPosition(request.Position)
+                },
+                cancellationToken);
+
+            return CrabEditClientResult<CrabSubaddressId>
+                .From(
+                    addCrabSubaddressResponse,
+                    identifier => new CrabSubaddressId(addCrabSubaddressResponse.AddressId));
+        }
+
+        public async Task<CrabEditResponse> Delete(
+            CrabIdToPersistentLocalIdItem addressId,
+            CancellationToken cancellationToken)
+        {
+            if (addressId?.HouseNumberId.HasValue ?? false)
+                return await _client.Delete(new RemoveHouseNumber { HouseNumberId = addressId.HouseNumberId.Value }, cancellationToken);
+
+            if (addressId?.SubaddressId.HasValue ?? false)
+                return await _client.Delete(new RemoveSubaddress { SubaddressId = addressId.SubaddressId.Value }, cancellationToken);
+
+            return CrabEditResponse.NothingExecuted;
+        }
+
+        public async Task<CrabEditClientResult> ChangeHouseNumber(
+            int houseNumberId,
+            ChangeAddressRequest request,
+            CancellationToken cancellationToken)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            var updateResponse = await _client.Update(
+                new EditHouseNumber
+                {
+                    HouseNumberId = houseNumberId,
+                    OfficiallyAssigned = request.OfficiallyAssigned,
+                    Position = MapPosition(request.Position),
+                    PostalCode = request.PostalCode.AsIdentifier().Map(IdentifierMappings.PostalCode),
+                    Status = request.Status.AsIdentifier().Map(IdentifierMappings.AddressStatus),
+                    IsCorrection = false,
+                },
+                cancellationToken);
+
+            return CrabEditClientResult.From(updateResponse);
+        }
+
+        public async Task<CrabEditClientResult> CorrectHouseNumber(
+            int houseNumberId,
+            CorrectAddressRequest request,
+            CancellationToken cancellationToken)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            var updateResponse = await _client.Update(
+                new EditHouseNumber
+                {
+                    HouseNumberId = houseNumberId,
+                    OfficiallyAssigned = request.OfficiallyAssigned,
+                    Position = MapPosition(request.Position),
+                    PostalCode = request.PostalCode.AsIdentifier().Map(IdentifierMappings.PostalCode),
+                    Status = request.Status.AsIdentifier().Map(IdentifierMappings.AddressStatus),
+                    IsCorrection = true,
+                },
+                cancellationToken);
+
+            return CrabEditClientResult.From(updateResponse);
+        }
```

Comment:
Can we turn this chunk into 1 call which simple differs between IsCorrection? 1 private method with all the code and the 2 publics just call it with true/false

Reply:
`ChangeAddressRequest ` and `CorrectAddressRequest` are different request models, you'd need a intermediate model to funnel it to 1 function. Change and Correct are 2 differnt business events in GRAR, they only happen to be exucuted with the same CRAB-edit function.